hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/ozone/scm/node/SCMNodeManager.java

```diff
@@ -580,21 +582,20 @@ public VersionResponse getVersion(SCMVersionRequestProto versionRequest) {
   @Override
   public SCMCommand register(DatanodeID datanodeID) {
 
-    SCMCommand errorCode = verifyDatanodeUUID(datanodeID);
-    if (errorCode != null) {
-      return errorCode;
+    SCMCommand responseCommand = verifyDatanodeUUID(datanodeID);
+    if (responseCommand != null) {
+      return responseCommand;
     }
-    DatanodeID newDatanodeID = new DatanodeID(UUID.randomUUID().toString(),
-        datanodeID);
-    nodes.put(newDatanodeID.getDatanodeUuid(), newDatanodeID);
+
+    nodes.put(datanodeID.getDatanodeUuid(), datanodeID);
```

Comment:
How do we handle the case when two datanodes attempt to register with the same datanodeID? I don't think we should allow both succeed without warning or error.